ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -246,17 +246,37 @@ static PullQueryResult executeOrRouteQuery(
   ) {
     List<List<?>> rows = null;
     if (node.isLocal()) {
-      LOG.debug("Query {} executed locally at host {} at timestamp {}.",
-                statement.getStatementText(), node.location(), System.currentTimeMillis());
-      pullQueryMetrics
-          .ifPresent(queryExecutorMetrics -> queryExecutorMetrics.recordLocalRequests(1));
-      rows = pullPhysicalPlan.execute(locations);
+      try {
+        LOG.debug("Query {} executed locally at host {} at timestamp {}.",
+                  statement.getStatementText(), node.location(), System.currentTimeMillis());
+        pullQueryMetrics
+            .ifPresent(queryExecutorMetrics -> queryExecutorMetrics.recordLocalRequests(1));
+        rows = pullPhysicalPlan.execute(locations);
+      } catch (Exception e) {
```

Comment:
It catches all because I don't know what kind of exceptions can be throw, it can be from materialization, from streams, from ksql. Also, It catches and then throws a `KsqlException` so what is the problem? We know that the exception has to do only with physical plan execution so it's not like it will catch something it shouldn't